processing/format_qaqc_driver.py

```diff
@@ -0,0 +1,430 @@
+                while process_id:
+                    d = self.db.trace_original_data_upload(
+                        self.conn,
+                        process_id)
+                    prior_process_id = d.get('prior_process_id')
+                    zip_process_id = d.get('zip_process_id')
+                    if prior_process_id:
+                        process_id = prior_process_id
+                    elif zip_process_id:
+                        process_id = zip_process_id
+                    else:
+                        uuid = d.get('upload_token')
+                        o_run_data = \
+                            self.db.get_latest_run_with_uuid(self.conn,
+                                                             uuid)
+                        if (o_run_data
+                                and
+                                (o_run_data
+                                 .get('process_timestamp')
+                                 > timestamp)):
+                            uuid = None
+                        break
+            if uuid and uuid not in rerun_uuids:
+                rerun_uuids.append(uuid)
+        return rerun_uuids
+
+    def get_new_upload_data(self,
+                            is_qaqc_processor=True,
+                            uuid=None,
+                            is_recovery=False):
+        if is_recovery:
+            new_data_upload_log = \
+                self.db.get_data_upload_log_with_uuid(
+                    self.conn,
+                    uuid)
+        else:
+            new_data_upload_log = \
+                self.db.get_new_data_upload_log(self.conn,
+                                                self.qaqc_processor_source,
+                                                is_qaqc_processor,
+                                                uuid)
+        log_ids_list = ' '.join([str(row.get('log_id'))
+                                 for row in new_data_upload_log])
+        if log_ids_list:
+            log_msg = f'Run with list of upload log ids: {log_ids_list} '
+            if uuid:
+                log_msg += f'with token: {uuid}'
+            if is_recovery:
+                log_msg = f'[RECOVERY MODE] {log_msg}'
+            else:
+                log_msg = f'[REGULAR MODE] {log_msg}'
+            _log.info(log_msg)
+        tasks = {}
+        for row in new_data_upload_log:
+            upload_id = row.get('log_id')
+            site_id = row.get('site_id')
+            token = row.get('upload_token')
+            if token not in self.blacklist_uuid:
+                zip_process_id = None
+                prior_process_id = None
+                run_type = 'o'
+                upload_comment = row.get('upload_comment', '')
+                if 'repair candidate for' in upload_comment:
+                    run_type = 'r'
+                    prior_process_id = upload_comment.split()[-1]
+                elif 'Archive upload for' in upload_comment:
```

Comment:
Probably a good idea to abstract these strings since you have it in multiple places.